tests/test_db.rs

```diff
@@ -408,8 +408,8 @@ fn test_get_updates_since_empty() {
     let path = DBPath::new("_rust_rocksdb_test_get_updates_since_empty");
     let db = DB::open_default(&path).unwrap();
     // get_updates_since() on an empty database
-    let mut iter = db.get_updates_since(0).unwrap();
-    assert!(iter.next().is_none());
+    let iter = db.get_updates_since(0).unwrap();
+    assert!(!iter.valid());
```

Comment:
Because of the GetUpdatesSince() fix from https://github.com/facebook/rocksdb/pull/9459,
the resulting iterator from DB::get_updates_since() will have its status Invalid on empty result.

Based on the [rocksdb code comment](https://github.com/facebook/rocksdb/blob/main/include/rocksdb/iterator.h#L65-L68), the iterator status is required to be `valid` when calling next().  As a result, the previous `assert!(iter.next().is_none())` will now crash instead.

The PR currently fixes the test, but please feel free to let me know if you prefer to check the iterator is valid inside the `next()` call instead.

Reply:
Yes, I think it makes sense to check an iterator for validity inside the `next` call. It is a safe api after all, so we don't want crashes here.

Reply:
Sounds good!  I've updated the PR!